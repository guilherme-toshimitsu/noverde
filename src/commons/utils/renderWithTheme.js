import React from 'react';
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../theme';
import { render } from '@testing-library/react';

function renderWithTheme(children) {
  return render(<ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>);
}

export { renderWithTheme };
