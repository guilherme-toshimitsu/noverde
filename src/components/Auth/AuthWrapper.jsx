import React, { useReducer } from 'react';
import PropTypes from 'prop-types';

import { createReducer } from '@utils/createReducer';
// import Api from '@api';
// import { useSnackBar } from '@components/SnackBar';

import * as actionTypes from './Reducer/authActionTypes';
import actions from './Reducer/authActions';
import initialState from './Reducer/authState';
import { AuthContextProvider } from './authContext';

const reducer = createReducer(actions);

const AuthWrapper = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  const doLogin = (cpf) => {
    dispatch({ type: actionTypes.LOGIN, cpf });
  };

  const doLogout = () => {
    dispatch({ type: actionTypes.LOGOUT });
  };

  const actionResolver = {
    doLogin: doLogin,
    doLogout: doLogout,
  };
  return (
    <AuthContextProvider value={[state, actionResolver]}>
      {children}
    </AuthContextProvider>
  );
};

AuthWrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

export default AuthWrapper;
