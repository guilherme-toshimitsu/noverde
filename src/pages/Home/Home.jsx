import React from 'react';
import PropTypes from 'prop-types';

import { FormCPF } from '@components/Forms';
import { StyledContainer, InsideContainer } from '@components/Container';

const Home = ({ history }) => {
  const onSuccess = () => history.push('/credit');
  return (
    <StyledContainer>
      <InsideContainer>
        <FormCPF onSuccess={onSuccess} />
      </InsideContainer>
    </StyledContainer>
  );
};

Home.propTypes = {
  history: PropTypes.object,
};

export default Home;
