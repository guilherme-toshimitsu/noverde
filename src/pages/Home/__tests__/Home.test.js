import React from 'react';
import { fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import AuthWrapper from '@components/Auth';

import Home from '../';
import { renderWithTheme } from '@commons/utils/renderWithTheme'
describe('<Home> Home Page Container', () => {
  it('renders', () => {
    const container = renderWithTheme(
    <>
      <AuthWrapper>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </AuthWrapper>
    </>
);
    expect(container).toMatchSnapshot();
  });

});
