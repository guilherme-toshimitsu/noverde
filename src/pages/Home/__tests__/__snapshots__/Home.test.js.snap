// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> Home Page Container renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iCoHVE jpsyLn"
        color="grey400"
      >
        <div
          class="sc-fujyUd jLtitf"
        >
          <h2
            class="sc-jSFkmK bUgUTJ"
          >
            Ola, queremos te ver #NOVERDE, para isso precisamos do seu CPF
          </h2>
          <div
            class="sc-gtssRu jQJSzM"
          >
            <input
              class="sc-eCApGN eSUYAQ"
              placeholder="Informe o seu CPF"
            />
          </div>
          <button
            class="sc-bdnylx kKDxdW"
            color="primary"
            data-testid="go-to-seller"
            disabled=""
            type="button"
          >
            PROXIMO
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iCoHVE jpsyLn"
      color="grey400"
    >
      <div
        class="sc-fujyUd jLtitf"
      >
        <h2
          class="sc-jSFkmK bUgUTJ"
        >
          Ola, queremos te ver #NOVERDE, para isso precisamos do seu CPF
        </h2>
        <div
          class="sc-gtssRu jQJSzM"
        >
          <input
            class="sc-eCApGN eSUYAQ"
            placeholder="Informe o seu CPF"
          />
        </div>
        <button
          class="sc-bdnylx kKDxdW"
          color="primary"
          data-testid="go-to-seller"
          disabled=""
          type="button"
        >
          PROXIMO
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
